### PR TITLE
[postgres] Use `DateConverter` for date type

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: lib/debezium/converters/time.go [hunk 1]
```diff
@@ -47,7 +46,14 @@ func (DateConverter) ToField(name string) debezium.Field {
 }
 
 func (DateConverter) Convert(value any) (any, error) {
-	return readerDebezium.ToDebeziumDate(value)
+	ts, isOk := value.(time.Time)
```
Comment (nathan-artie) [inline lib/debezium/converters/time.go:49]: Inlined `ToDebeziumDate` from https://github.com/artie-labs/reader/compare/nv/pg-date-converter?expand=1#diff-164927d903b5bb3e25f70e307b908bc755e3b09c752e8c7263f16d0fbeb001e5L8

File: lib/debezium/converters/time_test.go [hunk 2]
```diff
@@ -54,6 +54,12 @@ func TestDateConverter_Convert(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, 365, value)
 	}
+	{
+		// time.Time
```
Comment (nathan-artie) [inline lib/debezium/converters/time_test.go:58]: Moved test case from https://github.com/artie-labs/reader/pull/245/files#diff-6d97569992c2847cc9ab8549c03b32b729cf17d77184f972db7938466603bf6aL11

File: sources/postgres/adapter/fields.go [hunk 3]
```diff
@@ -87,11 +87,6 @@ func toDebeziumType(d schema.DataType) (Result, error) {
 			DebeziumType: string(debezium.Time),
 			Type:         "int32",
 		}, nil
-	case schema.Date:
```
Comment (nathan-artie) [inline sources/postgres/adapter/fields.go:90]: Compare with https://github.com/artie-labs/reader/blob/7a85861657e7e014a728bc3bfdcb7271aa2e7fc9/lib/debezium/converters/time.go#L56

File: lib/debezium/converters/time.go [hunk 4]
```diff
@@ -47,7 +46,14 @@ func (DateConverter) ToField(name string) debezium.Field {
...
+	if !isOk {
+		return 0, fmt.Errorf("object is not a time.Time object")
```
Comment (Tang8330) [inline lib/debezium/converters/time.go:51]: Does it have to return 0? Can we return nil?

File: lib/debezium/converters/time.go [hunk 5]
```diff
@@ -47,7 +46,14 @@ func (DateConverter) ToField(name string) debezium.Field {
...
+	}
+
+	unix := time.UnixMilli(0).In(time.UTC) // 1970-01-01
+	days := int(ts.Sub(unix).Hours() / 24)
```
Comment (Tang8330) [inline lib/debezium/converters/time.go:55]: nit: Can inline days

File: sources/postgres/adapter/values.go [hunk 6]
```diff
@@ -57,7 +48,7 @@ func ConvertValueToDebezium(col schema.Column, value any) (any, error) {
 	case schema.VariableNumeric:
 		encodedValue, err := debezium.EncodeDecimalToBase64(fmt.Sprint(value), debezium.GetScale(fmt.Sprint(value)))
 		if err != nil {
-			return util.SchemaEventPayload{}, fmt.Errorf("failed to encode decimal to b64 for key %s: %w", col.Name, err)
+			return nil, fmt.Errorf("failed to encode decimal to b64 for key %s: %w", col.Name, err)
```
Comment (nathan-artie) [inline sources/postgres/adapter/values.go:51]: No idea why we were returning `util.SchemaEventPayload{}` here.

Comment (Tang8330) [reply sources/postgres/adapter/values.go:51]: agreed

File: lib/debezium/converters/time.go (hunk 4 above)
Comment (nathan-artie) [reply lib/debezium/converters/time.go]: Yep, we can!

**Review by Tang8330 — APPROVED** [review]

**Pull request merged**